### PR TITLE
CHAOSPLT-315: Send disruption as payload in http notificaitions

### PR DESCRIPTION
**Pull request by diyarab**
## What does this PR do?

- [x] Adds new functionality
- [ ] Alters existing functionality
- [ ] Fixes a bug
- [ ] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- This PR adds disruption into http event notifier payload

## Code Quality Checklist

- [ ] The documentation is up to date.
- [ ] My code is sufficiently commented and passes continuous integration checks.
- [ ] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [ ] I leveraged continuous integration testing
    - [ ] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [x] I manually tested the following steps:
    - `x`
    - [x] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

File: eventnotifier/http/http.go [hunk 1]
```diff
@@ -126,27 +129,34 @@ func (n *Notifier) Notify(dis v1beta1.Disruption, event corev1.Event, notifType
...
 		TargetsCount:       len(dis.Status.TargetInjections),
 		Username:           emailAddr.Name,
 		UserEmail:          emailAddr.Address,
 	}
 
-	body, err := json.Marshal(notif)
-	if err != nil {
-		return fmt.Errorf("http notifier: couldn't send notification: %s", err.Error())
+	body := bytes.NewBuffer(nil)
+	if err := jsonapi.MarshalOnePayloadEmbedded(body, &notif); err != nil {
```
Comment (nathantournant) [inline eventnotifier/http/http.go:153]: Curious to know; what's the benefit of the `jsonapi` library in place of `json` ? Does it format things better for API calls ?

**Comment by ptnapoleon** [comment]
How/why is this different than #859 ?

**Comment by diyarab** [comment]
> How/why is this different than #859 ?

Pretty much the same, expect that I don't send timestamp yet

File: eventnotifier/http/http.go (hunk 1 above)
Comment (diyarab) [reply eventnotifier/http/http.go:153]: From what I understand, `jsonapi` structures responses to efficiently manage data relationships and minimize bandwidth, while plain `json` is simpler but lacks these optimizations. I think this article: [The Benefits of Using JSON API](https://nordicapis.com/the-benefits-of-using-json-api/) is a good read if you want a deeper explanation.

**Comment by ptnapoleon** [comment]
Shouldn't we just merge #859 instead, then?

**Pull request closed without merging**